### PR TITLE
gsocket

### PR DESCRIPTION
**Pull request by SkyperTHC**
Maintainer: skyper@thc.org
Compile tested: mips_24kc_musl 19.07.7
Run tested: mips_24kc_musl, 19.07.7, firmware builds successfully

Description: Global Socket allows two workstations on different private networks to communicate with each other. Through firewalls and through NAT - like there is no firewall.

### CODE REVIEW

File: net/gsocket/Makefile [hunk 1]
```diff
@@ -0,0 +1,71 @@
+# https://gsocket.io
```
Comment (BKPepe) [inline net/gsocket/Makefile:1]: Why this is commented?

File: net/gsocket/Makefile [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+include $(TOPDIR)/rules.mk
+
+PKG_NAME:=gsocket
+PKG_VERSION:=1.4.33
+PKG_RELEASE:=1
+
+#PKG_LICENSE:=BSD-2-Clause
+#PKG_LICENSE_FILES:=LICENSE
```
Comment (BKPepe) [inline net/gsocket/Makefile:10]: Why this is also commented?

File: net/gsocket/Makefile [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+PKG_BUILD_DIR:=$(BUILD_DIR)/$(PKG_NAME)-$(PKG_VERSION)
+
+# PKG_SOURCE_PROTO:=git
+# PKG_SOURCE_URL:=https://github.com/hackerschoice/gsocket.git
+# PKG_SOURCE_VERSION:=98a85e270c8ee5757134f00636a5358432cd3d83
```
Comment (BKPepe) [inline net/gsocket/Makefile:16]: This is leftover, I guess? Remove it as you download tarball from GitHub

File: net/gsocket/Makefile [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+include $(TOPDIR)/rules.mk
+
+PKG_NAME:=gsocket
+PKG_VERSION:=1.4.33
+PKG_RELEASE:=1
+
+#PKG_LICENSE:=BSD-2-Clause
+#PKG_LICENSE_FILES:=LICENSE
+
+PKG_BUILD_DIR:=$(BUILD_DIR)/$(PKG_NAME)-$(PKG_VERSION)
```
Comment (BKPepe) [inline net/gsocket/Makefile:12]: Isn't this default already? Why do you need to specify it here again?

File: net/gsocket/Makefile [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+# PKG_SOURCE_PROTO:=git
+# PKG_SOURCE_URL:=https://github.com/hackerschoice/gsocket.git
+# PKG_SOURCE_VERSION:=98a85e270c8ee5757134f00636a5358432cd3d83
+
+# Direct file download
+PKG_SOURCE_URL:=https://github.com/hackerschoice/gsocket/releases/download/v$(PKG_VERSION)/
+PKG_HASH:=5bbc850a274b933a4e8b0ac7d5bc8b0527c3eddbaee7f8a9389c284f27a6fe14
+
+# For testing copy to openwrt/dl
+PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.gz
```
Comment (BKPepe) [inline net/gsocket/Makefile:23]: If you will take a look at how others Makefiles are done, you will see there is order where this should be.

File: net/gsocket/Makefile [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+
+# dont need 'make install'
+#PKG_INSTALL:=1
+
+PKG_MAINTAINER:=skyper <skyper@thc.org>
```
Comment (BKPepe) [inline net/gsocket/Makefile:28]: Use your real name, otherwise this is not going to be accepted.

**Review by BKPepe — CHANGES_REQUESTED** [review]
1. This needs to be tested in OpenWrt master branch (compile and run tested)
Not against OpenWrt 19.07

2. You need to update your commit subject to something more exact.
Because commit subject only contains gsocket can be cryptic.
**Suggestion:** gsocket: add package can be much better or anything else. ;-)

3. You need to use your real name in the `Signed-off-by` and as well in the `PKG_MAINTAINER`

4. Fix the indentation in the install section

5. Why do we need to add this package to this repository? Add your reasoning to the commit message.

6. Add more detailed description in the Makefile.

7. Add `test.sh`, if possible. This ensures that this is going to be run tested by CI.

I hope you don't want to upstream this in this state because there are lots of comments, adding defaults. I suggest taking a look at how others Makefiles are done in this repository so that you can use it as an example.

**Comment by SkyperTHC** [comment]
thanks for all your helpful comments. I have made the changes (I hope) and I kindly ask you to check again.

**Comment by BKPepe** [comment]
> I forgot to answer your question 5 - why you should include it: 

You can add it to your commit message.

File: net/gsocket/Makefile [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+PKG_LICENSE_FILES:=LICENSE
+
+# PKG_BUILD_DIR:=$(BUILD_DIR)/$(PKG_NAME)-$(PKG_VERSION)
+
+# Direct file download
+PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.gz
+PKG_SOURCE_URL:=https://github.com/hackerschoice/gsocket/releases/download/v$(PKG_VERSION)/
+PKG_HASH:=5bbc850a274b933a4e8b0ac7d5bc8b0527c3eddbaee7f8a9389c284f27a6fe14
+
+PKG_MAINTAINER:=Ralf Kaiser <skyper@thc.org>
```
Comment (BKPepe) [inline net/gsocket/Makefile:23]:
```suggestion
PKG_NAME:=gsocket
PKG_VERSION:=1.4.33
PKG_RELEASE:=1

PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.gz
PKG_SOURCE_URL:=https://github.com/hackerschoice/gsocket/releases/download/v$(PKG_VERSION)/
PKG_HASH:=5bbc850a274b933a4e8b0ac7d5bc8b0527c3eddbaee7f8a9389c284f27a6fe14

PKG_MAINTAINER:=Ralf Kaiser <skyper@thc.org>
PKG_LICENSE:=BSD-2-Clause
PKG_LICENSE_FILES:=LICENSE
```

File: net/gsocket/Makefile [hunk 8]
```diff
@@ -0,0 +1,84 @@
+#
+# Copyright (C) 2015 OpenWrt.org
+#
+# This is free software, licensed under the GNU General Public License v2.
+# See /LICENSE for more information.
```
Comment (BKPepe) [inline net/gsocket/Makefile:5]: Omit or assign yourself.
According to [CONTRIBUTING](https://github.com/openwrt/packages/blob/8ece746b51f994ef74ca8ad20f5f05f370af2fbc/CONTRIBUTING.md#makefile-contents-should-contain).

File: net/gsocket/Makefile (hunk 7 above)
Comment (SkyperTHC) [reply net/gsocket/Makefile]: Ok. Thank you for your help. I've done those cahnges now.

Comment (BKPepe) [reply net/gsocket/Makefile]: These are different what I suggested. :(

File: net/gsocket/Makefile [hunk 9]
```diff
@@ -0,0 +1,75 @@
+include $(TOPDIR)/rules.mk
+
+PKG_NAME:=gsocket
+PKG_VERSION:=1.4.33
+PKG_RELEASE:=1
+
+PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.gz
+
+PKG_SOURCE_URL:=https://github.com/hackerschoice/gsocket/releases/download/v$(PKG_VERSION)/
```
Comment (BKPepe) [inline net/gsocket/Makefile:9]:
```suggestion
PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.gz
PKG_SOURCE_URL:=https://github.com/hackerschoice/gsocket/releases/download/v$(PKG_VERSION)/
```

**Comment by BKPepe** [comment]
What's left is to create file ``test.sh`` to be run tested by CI. How it looks, you can find in this repository to take a look how it is done for other packages. Not all of them has it and then you can squash your commits, so the formalities test pass.

**Comment by SkyperTHC** [comment]
> What's left is to create file `test.sh` to be run tested by CI. How it looks, you can find in this repository to take a look how it is done for other packages. Not all of them has it and then you can squash your commits, so the formalities test pass.

that's a good tip. I've done this now. At least it checks that the binary executes and has the right version. Thanks for helping. I also remove the empty line in the Makefile.

**Review by neheb — APPROVED** [review]

**Comment by SkyperTHC** [comment]
Hi Josef,

The mips_24kc target failed for my openwrt app. See
https://github.com/openwrt/packages/runs/3103122951?check_suite_focus=true.
It compiles fine on my system. Can you point me into the right direction
how I can this fixed?

regards,

ralf

**Review by BKPepe — APPROVED** [review]

**Comment by BKPepe** [comment]
There is one thing before merging this and that's the failing test for formalities. By adding ``.patch`` to the URL.
``https://github.com/openwrt/packages/pull/16103.patch``, I can see that the author of the commit is:
> From: rootTHC <root@thc.org>

Can you fix it? It should be possible to fix it by using following command:
``git commit --amend --reset-author`` if you configured your **Git config**. If not then you can use `--author="Foo <foo@foo.tld>"`

**Comment by SkyperTHC** [comment]
Hi,

hmm. it does say "Signed-off-by: Ralf Kaiser ***@***.***>" but it also
says "From: rootTHC ***@***.***". I thought the 'Singed-off-by' was
sufficient to pass the test.

i've done now as you instructed.

thanks for helping,

ralf

**Comment by SkyperTHC** [comment]
Hi Josef,

I do not understand what  you like me to do differently. I'm doing this:


*git commit --amend --author="Ralf Kaiser ***@***.*** ***@***.***>>"
--no-edit --signoff -m "gsocket: add package"git push --force-with-lease
origin master*

Can you explain to me what's not working? Then I can take a closer look at
it.

thanks,

Ralf

**Comment by BKPepe** [comment]
Well, you are ammending changes from your stash to the single commit. There are many ways, how you can do it.

I will show you just one:
```
git rebase -i HEAD~3
```

And then mark these two recent commits as fixups, save it and those commits would be merged to the single one. So, what you need to do is to force push it.

**Comment by BKPepe** [comment]
Closing this for now.

**Pull request closed without merging**